test/dotnet-build.Tests/BuildProjectToProjectTests.cs

```diff
@@ -15,26 +16,35 @@ public class ProjectToProjectDependenciesIncrementalTest : IncrementalTestBase
     {
         private string[] _projects = new[] { "L0", "L11", "L12", "L21", "L22" };
 
-        public ProjectToProjectDependenciesIncrementalTest() : base(
-            Path.Combine(AppContext.BaseDirectory, "TestAssets", "TestProjects", "TestProjectToProjectDependencies"),
-            "L0",
-            "L0 L11 L12 L22 L21 L12 L22 " + Environment.NewLine)
+        private string MainProjectExe
         {
+            get
+            {
+                return MainProject + (RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? ".exe" : "");
+            }
+        }
+
+        public ProjectToProjectDependenciesIncrementalTest()
+        {
+            MainProject = "L0";
+            ExpectedOutput = "L0 L11 L12 L22 L21 L12 L22 " + Environment.NewLine;
+
         }
 
         [Theory,
-        InlineData("L0", new[] { "L0" }),
-        InlineData("L11", new[] { "L0", "L11" }),
-        InlineData("L12", new[] { "L0", "L11", "L12" }),
-        InlineData("L22", new[] { "L0", "L11", "L12", "L22" }),
-        InlineData("L21", new[] { "L0", "L11", "L21" })
+        InlineData("1", "L0", new[] { "L0" }),
+        InlineData("2", "L11", new[] { "L0", "L11" }),
+        InlineData("3", "L12", new[] { "L0", "L11", "L12" }),
+        InlineData("4", "L22", new[] { "L0", "L11", "L12", "L22" }),
+        InlineData("5", "L21", new[] { "L0", "L11", "L21" })
         ]
-        public void TestIncrementalBuildOfDependencyGraph(string projectToTouch, string[] expectedRebuiltProjects)
+        public void TestIncrementalBuildOfDependencyGraph(string testIdentifer, string projectToTouch, string[] expectedRebuiltProjects)
```

Comment:
It is not required. But if you don't specify the `identifier` the copy of test asset will get overwritten. I initially thought of using timestamps but then it is going to be difficult to identify a particular theory's instance of the test asset.